apps/backend/src/lib/request-checks.tsx

```diff
@@ -105,6 +105,67 @@ export async function ensureUserHasTeamPermission(
   }
 }
 
+export async function ensureUserTeamPermissionExist(
```

Comment:
I think this function is a little bit duplicated to `ensureUserHasTeamPermission`

Reply:
I am combining these functions, but logically, they signify different types of errors.

1. `ensureUserHasTeamPermission`
   - This function throws a `TeamPermissionRequired` error, which indicates an authorization error. Users might misconstrue this error as a permission issue in the case of the **Revoke a team permission from a user** API request.

2. `ensureUserTeamPermissionExist`
   - This function throws a `UserTeamPermissionNotFound` error, indicating that the permission the user is trying to delete does not exist. This is more appropriate for the **Revoke a team permission from a user** API request.

Reply:
yeah, I added `errorType` parameter to decide which type of error to throw in the `ensureUserHasTeamPermission` function.